File: Digito Verificador/Modulo 10 Padrao/Teste - Arquivo 10/GerarListaMatriculasMod10.py
def formata_MATRICULA(MATRICULA):
    return f"{MATRICULA:06d}"

def mod10(MATRICULA):
    SOMA = 0
    for i in range(6):
        DIGITO = int(MATRICULA[i])
        if i % 2 != 0 and DIGITO * 2 > 9:
            SOMA += DIGITO * 2 - 9
        elif i % 2 != 0 and DIGITO * 2 <= 9:
            SOMA += DIGITO * 2
        else:
            SOMA += DIGITO
    DV = SOMA % 10
    if DV != 0:
        DV = 10 - DV
    return DV

def main():
    try:
        ARQUIVO = open("numeros.txt", "w")

        print("GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM \"DV\"")
        print("Padrao: Modulo 10")
        print("---------------------------------------------------------------")
        print()
        print("Aguarde ARQUIVO sendo gerado... ", end="", flush=True)

        for I in range(1000000):
            MATRICULA = formata_MATRICULA(I)
            DV = mod10(MATRICULA)
            ARQUIVO.write(f"{MATRICULA[:3]}.{MATRICULA[3:]}-{DV}\n")

        ARQUIVO.close()

        print("\n\nSUCESSO: Arquivo gerado.\n")

    except IOError:
        print("Erro ao abrir o ARQUIVO.")
        print("Programa encerrado.")

if __name__ == "__main__":
    main()

File: Digito Verificador/Modulo 10 Padrao/Teste - Arquivo 10/GerarListaMatriculasMod10.scala
import java.io.PrintWriter

object Main {
  def formataMatricula(MATRICULA: Int): String = {
    f"$MATRICULA%06d"
  }

  def mod10(MATRICULA: String): Int = {
    var SOMA = 0
    for (i <- 0 until 6) {
      val DIGITO = MATRICULA(i) - '0'
      if (i % 2 != 0 && DIGITO * 2 > 9)
        SOMA += DIGITO * 2 - 9
      else if (i % 2 != 0 && DIGITO * 2 <= 9)
        SOMA += DIGITO * 2
      else
        SOMA += DIGITO
    }
    val DV = if (SOMA % 10 == 0) 0 else 10 - SOMA % 10
    DV
  }

  def main(args: Array[String]): Unit = {
    val ARQUIVO = new PrintWriter("numeros.txt")

    println("GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM \"DV\"")
    println("Padrao: Modulo 10")
    println("---------------------------------------------------------------")
    println()

    if (ARQUIVO == null) {
      Console.err.println("Erro ao abrir o ARQUIVO.")
      println("Programa encerrado.")
      return
    }

    print("Aguarde ARQUIVO sendo gerado... ")

    for (I <- 0 to 999999) {
      val MATRICULA = formataMatricula(I)
      val DV = mod10(MATRICULA)
      ARQUIVO.println(s"${MATRICULA.substring(0, 3)}.${MATRICULA.substring(3)}-$DV")
    }

    ARQUIVO.close()

    println("\n\nSUCESSO: Arquivo gerado.\n")
  }
}
